src/components/MediaQuery.js

```
@@ -0,0 +1,97 @@
+import Base from '../abstracts/Base';
+import useResize from '../services/resize';
+import nextFrame from '../utils/nextFrame';
+
+/**
+ * MediaQuery component.
+ *
+ * <div data-component="MediaQuery" data-active-breakpoints="l xl">
+ *   <div data-component="Foo"></div>
+ * </div>
+ */
+export default class MediaQuery extends Base {
+  /**
+   * Component's configuration.
+   *
+   * @return {Object}
+   */
+  get config() {
+    return {
+      name: 'MediaQuery',
+    };
+  }
+
+  /**
+   * Mounted hook.
+   */
+  mounted() {
+    this.test();
+    nextFrame(() => this.test());
+  }
+
+  /**
+   * Resized hook.
+   */
+  resized() {
+    this.test();
+  }
+
+  /**
+   * Get the first element child of the component, as it must be another Base component that could
+   * be either $mounted or $destroyed.
```

This seems like a specific behavoir, maybe we could throw an error bellow to explain this

---

Added in ef616a3 😉